ui/components/ui/deprecated-networks/deprecated-networks.js

```diff
@@ -1,38 +1,78 @@
   BackgroundColor,
   BorderRadius,
   Severity,
 } from '../../../helpers/constants/design-system';
 
-import { getCurrentChainId } from '../../../selectors';
+import { getCurrentNetwork } from '../../../selectors';
 import { getCompletedOnboarding } from '../../../ducks/metamask/metamask';
-import { BannerAlert, Box, ButtonLink } from '../../component-library';
+import { BannerAlert, Box } from '../../component-library';
 import {
+  AURORA_DISPLAY_NAME,
   CHAIN_IDS,
+  CURRENCY_SYMBOLS,
   DEPRECATED_NETWORKS,
+  NEAR_AURORA_MAINNET_IMAGE_URL,
 } from '../../../../shared/constants/network';
+import { editAndSetNetworkConfiguration } from '../../../store/actions';
+import { MetaMetricsNetworkEventSource } from '../../../../shared/constants/metametrics';
 
 export default function DeprecatedNetworks() {
-  const currentChainID = useSelector(getCurrentChainId);
-  const [isShowingWarning, setIsShowingWarning] = useState(false);
+  const { id, chainId, rpcUrl } = useSelector(getCurrentNetwork);
+  const [isClosed, setIsClosed] = useState(false);
   const completedOnboarding = useSelector(getCompletedOnboarding);
   const t = useI18nContext();
+  const dispatch = useDispatch();
 
-  useEffect(() => {
-    if (completedOnboarding && DEPRECATED_NETWORKS.includes(currentChainID)) {
-      setIsShowingWarning(true);
-    } else {
-      setIsShowingWarning(false);
-    }
-  }, [currentChainID, completedOnboarding]);
+  if (!completedOnboarding || isClosed) {
+    return null;
+  }
 
-  const { bannerAlertDescription, actionBtnLinkURL } =
-    getDeprecationWarningCopy(t, currentChainID);
+  let props;
+  if (chainId === CHAIN_IDS.GOERLI) {
+    props = {
+      description: t('deprecatedGoerliNtwrkMsg'),
+      actionButtonLabel: t('learnMoreUpperCase'),
+      actionButtonProps: {
+        href: 'https://github.com/eth-clients/goerli#goerli-goerlitzer-testnet',
+        externalLink: true,
+      },
+    };
+  } else if (DEPRECATED_NETWORKS.includes(chainId)) {
+    props = { description: t('deprecatedNetwork') };
+  } else if (
+    chainId === CHAIN_IDS.AURORA &&
+    rpcUrl.startsWith('https://aurora-mainnet.infura.io/')
+  ) {
+    props = {
+      description: t('auroraRpcDeprecationMessage'),
+      actionButtonLabel: t('switchToNetwork', ['mainnet.aurora.dev']),
+      actionButtonOnClick: async () => {
+        setIsClosed(true);
+        await dispatch(
+          editAndSetNetworkConfiguration(
+            {
+              networkConfigurationId: id,
+              chainId: CHAIN_IDS.AURORA,
+              nickname: AURORA_DISPLAY_NAME,
+              rpcUrl: 'https://mainnet.aurora.dev',
+              ticker: CURRENCY_SYMBOLS.ETH,
+              rpcPrefs: {
+                imageUrl: NEAR_AURORA_MAINNET_IMAGE_URL,
+                blockExplorerUrl: 'https://aurorascan.dev',
+              },
+            },
+            { source: MetaMetricsNetworkEventSource.DeprecatedNetworkModal },
+          ),
+        );
+      },
+    };
+  }
```

Comment:
Great work !. to ensure robustness and catch any potential issues early, I recommend adding unit tests covering various if/else scenarios.

Reply:
I added a storybook for aurora

<img width="528" alt="Screenshot 2024-05-15 at 5 06 21 PM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/52a50826-8d8b-4dd0-8aa4-aa696fc1d201">